website/docs/batch-execution.mdx

```diff
@@ -0,0 +1,130 @@
+---
+id: batch-execution
+title: Batch execution
+description: Batch multiple GraphQL requests into a single operation
+---
+
+Batch execution is a technique for consolidating multiple operations that target a single schema. Rather than executing each operation individually, all operations can be combined and executed as one.
+
+For example, given the following GraphQL operations:
+
+```graphql
+query {
+  field1
+  field3(input: "A")
+}
+query {
+  tricky: field2
+  field3(input: "B")
+}
+```
+
+These can be merged into one operation, and the resulting data can be unpacked into the original shape of the multiple requests:
+
+```graphql
+query {
+  _0_field1: field1
+  _0_field3: field3(input: "A")
+  _1_tricky: field2
+  _1_field3: field3(input: "B")
+}
+```
+
+**Batch execution is useful because:**
+
+- Multiple operations can be combined into one network request when targeting remote services.
+- Combined operations are guarenteed to multiplex, even with servers that execute incoming requests serially.
+- Smaller and more granular GraphQL queries may be composed and cached individually, and then batched. This offers a strategy for sub-request caching.
+
+> _Note:_ an alternative batching pattern transmits multiple operations [as an array](https://www.apollographql.com/blog/apollo-client/performance/batching-client-graphql-queries/#5853), and requires services to be specially configured for array inputs. 
+GraphQL Tools uses the plain GraphQL approach [devised by Gatsby](https://github.com/gatsbyjs/gatsby/pull/22347), and is compatible with any standard GraphQL service.
+
+## Build an executor
+
+An executor is a generic function that wraps a GraphQL execution with the following interface:
+
+```js
+type Executor = (request: Request) => Promise<ExecutionResult>;
+
+type Request = {
```

Comment:
The type name changed in v8 I think to ExecutionRequest, on this line and scattered about.

Reply:
Fixed in 1642bdd9e67243d742bbb36e32207009eda61ee0.